iceoryx_posh/source/roudi/roudi_cmd_line_parser.cpp

```diff
@@ -24,7 +24,7 @@ namespace iox
 {
 namespace config
 {
-void CmdLineParser::parse(int argc, char* argv[], const CmdLineArgumentParsingMode cmdLineParsingMode) noexcept
+CmdLineArgs_t CmdLineParser::parse(int argc, char* argv[], const CmdLineArgumentParsingMode cmdLineParsingMode) noexcept
```

Comment:
I'm wondering if this should be a `cxx::expected<CmdLineArgs_t>`? But we would only partially get rid of the `run` member, since it is still needed for `--help`

Reply:
I had the same though last week. However, I came to the conclusion a parser taking `argc` and `argv` can not fail. If the args are not supported we inform the user and return a defaulted `CmdLineArgs_t`.

Reply:
well, wrong args result in RouDi not running. I would consider this as fail ;)

Reply:
When writing the test I realised it's better to switch to an `expected`. We should write more tests :P